multiqc/modules/ivar/ivar.py

```diff
@@ -0,0 +1,167 @@
+        self.ivar_data = dict()
+        self.parsed_primers = dict()
+        for f in self.find_log_files('ivar/trim', filehandles=True):
+            parsed_data = self.parse_ivar(f)
+            if parsed_data is not None and len(parsed_data) > 0:
+                self.ivar_data[f['s_name']] = parsed_data
+                self.add_data_source(f, f['s_name'])
+        for f in self.find_log_files('ivar/trim'):
+            parsed_primers = self.parse_ivar_primer_stats(f)
+            if parsed_primers is not None and len(parsed_primers) > 0:
+                self.parsed_primers[f['s_name']] = parsed_primers
+                self.add_data_source(f, f['s_name'])
+
+        # Filter to strip out ignored sample names
+        self.ivar_data = self.ignore_samples(self.ivar_data)
+
+        # Warning when no files are found
+        if len(self.ivar_data) == 0:
+            raise UserWarning
+
+        # Write parsed data to a file
+        self.write_data_file(self.ivar_data, 'multiqc_ivar_summary')
+        
+        #Primers too
+        self.write_data_file(self.parsed_primers, 'multiqc_ivar_primers')
+        
+        #Found reports or not?
+        log.info("Found {} reports".format(len(self.ivar_data)))
+
+        # Basic Stats Table
+        self.ivar_general_stats_table()
+
+        #Heatmap info
+        self.status_heatmap()
+        
+    # Parse an ivar report
+    def parse_ivar(self, f):
+        parsed_data = dict()
+        regexes = {
+            'mapped_reads': r'(?:Found\s)(\d+)(?:\smapped)',
+            'total_reads': r'(?:Trimmed primers from )(?:\d+\.\d+\% \()?(\d+)',
+            'reads_outside_primer_region': r'^(?:\d+\.\d+\% \()?(\d+)(?:\))?(?:.*[of]?)reads\s(?:that\s)?started',
+            'reads_too_short_after_trimming': r'^(?:\d+\.\d+\% \()?(\d+)(?:\))?(?:.*[of]?)reads\swere(?: quality trimmed | shortened)'
+        }
+        for l in f['f']:
+            # Search regexes for stats
+            for k, r in regexes.items():
+                match = re.search(r, l)
+                if match:
+                    parsed_data[k] = int(match.group(1))
+        
+        return parsed_data
+
+    #Parse Primer stats appropriately
+    def parse_ivar_primer_stats(self, f):
+        primers = OrderedDict()
+        regex = "^(.*)(?:\t+)(\d+$)"
+        # Search regexes for stats
+        for l in f['f'].splitlines():
+            match = re.search(regex, l)
+            if match:
+                primer = match.group(1)
+                counts = int(match.group(2))
+                primers[primer] = counts
+        log.info("Found {} primers".format(len(primers)))
+        return primers
+
+    # Add to general stats table
+
+    def ivar_general_stats_table(self):
+        """ Take the parsed stats from the iVAR report and add it to the
+        basic stats table"""
+
+        headers = OrderedDict()
+        headers['mapped_reads'] = {
+            'title': 'Total mapped reads',
```

Comment:
Please see https://multiqc.info/docs/#number-base-multiplier

Also `shared_key`: `read_counts` https://multiqc.info/docs/#step-3-adding-to-the-general-statistics-table